torch/distributed/tensor/placement_types.py

```diff
@@ -134,6 +117,11 @@ def _local_shard_size_on_dim(
     ) -> tuple[int, int]:
         """
         returns the local shard size and offset on a given tensor dim
+
+        # TODO(whc)
+         - `size_on_dim` arg has a pretty confusing name, would 'global_dim_size' be better?
```

Comment:
`global_dim_size` implies the dim size of the global tensor. However, the `size_on_dim` is the local dim size before being further sharded. Think of 2D sharding, the `size_on_dim` when sharding on FSDP is `global_dim_size // tp_size`.

Reply:
moved these changes to a new PR on the stack and removed the TODOs from here